### PR TITLE
[DUOS-2233][risk=no] Add broad dataset catalog page

### PR DESCRIPTION
**Pull request by connorlbark**
### Addresses

https://broadworkbench.atlassian.net/browse/DUOS-2233

Basically, Jonathan wanted to have a dataset catalog page which is specific to an institution/DAC. It would have their primary color, logo, and by default only show their datasets. This is an 'experiment' to see what this would look like for Broad DACs, with the ability to very easily expand to new DACs.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/Routes.js [hunk 1]
```diff
@@ -90,6 +91,7 @@ const Routes = (props) => (
     <AuthenticatedRoute path="/admin_manage_lc/" component={AdminManageLC} props={props} rolesAllowed={[USER_ROLES.admin]} />
     <AuthenticatedRoute path="/admin_manage_dar_collections/" component={AdminManageDarCollections} props={props} rolesAllowed={[USER_ROLES.admin]} />
     <AuthenticatedRoute path="/dataset_catalog" component={DatasetCatalog} props={props} rolesAllowed={[USER_ROLES.admin, USER_ROLES.all]} />
```
Comment (rushtong) [inline src/Routes.js:93]: Not for this PR, but `[USER_ROLES.admin, USER_ROLES.all]` got me wondering about what does `all` mean and why is admin not included in `all`, if indeed it isn't.

**Review by rushtong — APPROVED** [review]
I really like this approach 👍🏽  This works as I believe it is intended to. This PR is good to go as is, in my opinion. One thing I would like to sort out is the ability to re-navigate to the custom page, but that is for a future discussion/task.

**Review by quazi-h — APPROVED** [review]
Changes look good to me. Running locally the Dataset Catalog on the DAC Console is running smoothly and filtering correctly. Does this overlap with the work in [DUOS-1929](https://broadworkbench.atlassian.net/browse/DUOS-1929)? It seems like this might be meant to replace it.

**Comment by connorlbark** [comment]
> Changes look good to me. Running locally the Dataset Catalog on the DAC Console is running smoothly and filtering correctly. Does this overlap with the work in [DUOS-1929](https://broadworkbench.atlassian.net/browse/DUOS-1929)? It seems like this might be meant to replace it.

This is more meant to go along side it, where the DAC could send this link to researchers. So this is more oriented for researchers, whereas that is more meant for DAC chairs.

[DUOS-1929]: https://broadworkbench.atlassian.net/browse/DUOS-1929?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Pull request merged**